internal/pkg/argocd/argocd.go

```diff
@@ -144,7 +143,7 @@ func getEnv(key, fallback string) string {
 	return fallback
 }
 
-func createArgoCdClient() (apiclient.Client, error) {
+func createArgoCdClients() (appClient application.ApplicationServiceClient, projClient projectpkg.ProjectServiceClient, settingClient settings.SettingsServiceClient, err error) {
```

Comment:
Just a suggestion, make a struct for the various clients, that way you're only returning a single struct that contains clientStruct.app, clientStruct.project, client.settingClient, etc. Would also make it easier later if you need to add another client, you wouldn't need to modify the func signature.